packages/eas-cli/src/commands/fingerprint/compare.ts

```diff
@@ -170,6 +174,186 @@ export default class FingerprintCompare extends EasCommand {
+    ...optionsFromWorkflow,
+    platforms: firstFingerprintPlatforms.map(appPlatformToString),
+    debug: true,
+    env: undefined,
+  });
+  if (!projectFingerprint) {
+    throw new Error('Project fingerprints can only be computed for projects with SDK 52 or higher');
+  }
+
+  const uploadedFingerprint = await maybeUploadFingerprintAsync({
+    hash: projectFingerprint.hash,
+    fingerprint: {
+      fingerprintSources: projectFingerprint.sources,
+      isDebugFingerprintSource: Log.isDebug,
+    },
+    graphqlClient,
+  });
+  await FingerprintMutation.createFingerprintAsync(graphqlClient, projectId, {
+    hash: uploadedFingerprint.hash,
+    source: uploadedFingerprint.fingerprintSource,
+  });
+
+  return { fingerprint: projectFingerprint, origin: 'project' };
+}
+
+async function getFingerprintFragmentFromHashAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  hash: string
+): Promise<FingerprintFragment> {
+  const displayName = await getDisplayNameForProjectIdAsync(graphqlClient, projectId);
+  const fingerprint = await FingerprintQuery.byHashAsync(graphqlClient, {
+    appId: projectId,
+    hash,
+  });
+  if (!fingerprint) {
+    throw new Error(`Fingerprint with hash ${hash} was not uploaded for ${displayName}.`);
+  }
+  return fingerprint;
+}
+
+async function getFingerprintAsync(fingerprintFragment: FingerprintFragment): Promise<Fingerprint> {
+  const fingerprintDebugUrl = fingerprintFragment.debugInfoUrl;
+  if (!fingerprintDebugUrl) {
+    throw new Error(
+      `The source for fingerprint hash ${fingerprintFragment.hash} was not computed.`
+    );
+  }
+  const fingerprintResponse = await fetch(fingerprintDebugUrl);
+  return (await fingerprintResponse.json()) as Fingerprint;
+}
+
+function getFingerprintOptionsFromWorkflow(
+  platforms: AppPlatform[],
+  workflowsByPlatform: Record<Platform, Workflow>
+): { workflow?: Workflow; ignorePaths?: string[] } {
+  if (platforms.length === 0) {
+    throw new Error('Could not determine platform from fingerprint sources');
+  }
+
+  // Single platform case
+  if (platforms.length === 1) {
+    const platform = platforms[0];
+    return { workflow: workflowsByPlatform[appPlatformToPlatform(platform)] };
+  }
+
+  // Multiple platforms case
+  const workflows = platforms.map(platform => workflowsByPlatform[appPlatformToPlatform(platform)]);
+
+  // If all workflows are the same, return the common workflow
+  const [firstWorkflow, ...restWorkflows] = workflows;
+  if (restWorkflows.every(workflow => workflow === firstWorkflow)) {
+    return { workflow: firstWorkflow };
+  }
+
+  // Generate ignorePaths for mixed workflows
+  const ignorePaths = platforms
+    .filter(platform => workflowsByPlatform[appPlatformToPlatform(platform)] === Workflow.MANAGED)
+    .map(platform => `${appPlatformToString(platform)}/**/*`);
+
+  return { ignorePaths };
+}
+
+function inferPlatformsFromSource(fingerprint: Fingerprint): AppPlatform[] {
```

Comment:
talked about this offline, but I think if we can avoid this by limiting args or splitting out commands that'd be preferable. The main issue is that (I believe) it's possible to create a fingerprint for either platform or both that doesn't have either string being checked for (by using a specially-crafted .fingerprintIgnore or sourceSkips).